File: backend/service/local_user.py
from backend.repository.db import DbState
from backend.repository import local_user as local_user_repo


class LocalUserService:
    def __init__(self, db: DbState):
        self.db = db

    def get_local_users(self) -> str:
        return local_user_repo.get_local_users(self.db)

    def save_local_users(self, accounts_json: str) -> None:
        local_user_repo.save_local_users(self.db, accounts_json)

File: src-tauri/src/service/local_user.rs
use crate::repository::{local_user as repo, DbState};

pub fn get_local_users(db: &DbState) -> Result<String, String> {
    repo::get_local_users(db)
}

pub fn save_local_users(db: &DbState, accounts_json: &str) -> Result<bool, String> {
    repo::save_local_users(db, accounts_json).map(|_| true)
}
